### PR TITLE
fix early glob expansion, starlark patterns

### LINKED ISSUE #117: [Bug]: Starlark formatter no longer touches BUILD files
### What happened?

Looks like when format.sh was overhauled, rules_lint stopped getting BUILD/BUILD.bazel etc files outside of the root. the patterns for starlark at least are incomplete, and probably for other languages, too:

```console
$ git ls-files --cached --modified --other --exclude-standard BUILD.bazel
BUILD.bazel
$ find . -name BUILD.bazel | wc -l
737
```

looks like a little additional processing is needed from linguist to add '**/<PATTERN>' for any pattern that has no wildcards in it already

### Version

Development (host) and target OS/architectures: all

Output of `bazel --version`: bazel 7.0.0

Version of the Aspect rules, or other relevant rules from your
`WORKSPACE` or `MODULE.bazel` file:

Language(s) and/or frameworks involved: all

### How to reproduce

```shell
make an edit to a BUILD.bazel file in a subdirectory that would be corrected by buildifier. Run the formatter. Note it was not corrected.
```

### Any other information?

additionally, if you trace execution of ls-files in format.sh it looks like we're getting some unexpected wildcard handling:

```
...
++ case "$language" in
++ patterns=('BUCK' 'BUILD' 'BUILD.bazel' 'MODULE.bazel' 'Tiltfile' 'WORKSPACE' 'WORKSPACE.bazel' '*.bzl' '*.star')
++ '[' 0 -eq 0 ']'
++ git ls-files --cached --modified --other --exclude-standard BUCK BUILD BUILD.bazel MODULE.bazel Tiltfile WORKSPACE WORKSPACE.bazel extensions.bzl repositories.bzl '*.star'
...
```

notable here is the expansion of *.bzl into the two matching files in the root of my repository. it looks like this is because of an unquoted bash array expansion- might want to turn on shellcheck from rules_lint in rules_lint :)

### PR DESCRIPTION
**Pull request by pjjw**
tl;dr this is a partial correction of #117 

Looks like the patterns in the reworked format.sh are not catching everything anymore, partially because of an unquoted bash array expansion, and partially because the patterns are incomplete as written here.

This is an incomplete fix, so don't expect this to be merged exactly (don't have more time today), but at least wanted to get this reported and put up this fix for at least the part i'm concerned about

More work is probably needed for other languages, as the patterns look to be probably deficient in the same way the starlark patterns are
---

### Type of change

- Bug fix

### CODE REVIEW

**Comment by pjjw** [comment]
obsoleted by #120
